### PR TITLE
Move Swift.php and start with travis tests

### LINKED ISSUE #1: Move swift primary storage to an app of its own
## Current Situation
the swift implementation lives in core https://github.com/owncloud/core/blob/master/lib/private/Files/ObjectStore/Swift.php and relies on rackspace libraries which are for sure outdated.

## Todo
- [x] move swift.php into this repo
- [x] move the continuous integration in here https://github.com/owncloud/core/blob/master/Jenkinsfile#L115-L127
- [ ] require new rackspace library - mind the guzzle gap
- [ ] add makefile to allow app bundling
- [ ] create signature
- [ ] upload to marketplace
- [ ] remove from master
- [ ] remove from stable10

## Motivation
- allow own release cycle via marketplace
- shrink core
- reduce ci time

### PR DESCRIPTION
**Pull request by mrow4a**
@DeepDiver1975 

fixes #1

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -27,6 +27,13 @@ before_install:
...
+  # Start ceph
+  - export DEBUG=on
+  - bash apps/files_primary_swift/tests/env/start-swift-ceph.sh
+  #- cp tests/objectstore/swift.config.php config/autotest-storage-swift.config.php
+  #- cp apps/objectstore/tests/env/autotest.sh autotest.sh
+  - echo "memory_limit=2048M" >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini\
+
 script:
   # Run phpunit tests
   - NOCOVERAGE=1 RUN_OBJECTSTORE_TESTS=1 PRIMARY_STORAGE_CONFIG="swift" ./autotest.sh sqlite
```
Comment (DeepDiver1975) [inline .travis.yml]: try to call make - https://github.com/owncloud/objectstore/blob/master/.travis.yml#L49

File: tests/env/start-swift-ceph.sh [hunk 2]
```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
```
Comment (DeepDiver1975) [inline tests/env/start-swift-ceph.sh]: is this file a copy of https://github.com/owncloud/objectstore/blob/master/tests/env/start-s3-ceph.sh ?

Comment (mrow4a) [reply tests/env/start-swift-ceph.sh]: Yes, and this is the one which actually started working without `Error: kernel module loop not loaded. Needed by docker image xenopathic/ceph-keystone`, lets see if this is correct.

**Comment by mrow4a** [comment]
@tboerger can you help here? I cannot figure out what is wrong with swift travis-ci. I also have big problems running it locally with docker.  You seem the last resort, last line of defence ;/

File: README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Primary storage SWIFT
+:construction: Master Ci status: [![Build Status](https://magnum.travis-ci.com/owncloud/files_primary_swift.svg?token=q8ZoWBCat8DFpZ2ALfXP&branch=master)](https://magnum.travis-ci.com/owncloud/files_primary_swift)
```
Comment (tboerger) [inline README.md:2]: A token within a public repo?

**Comment by tboerger** [comment]
> @tboerger can you help here? I cannot figure out what is wrong with swift travis-ci. I also have big problems running it locally with docker. You seem the last resort, last line of defence ;/

What should I help here?

File: tests/env/start-swift-ceph.sh (hunk 2 above)
Comment (DeepDiver1975) [reply tests/env/start-swift-ceph.sh]: Please compare the .travis.yaml files to ensure the same vms are used

Comment (mrow4a) [reply tests/env/start-swift-ceph.sh]: what do you mean?

Comment (DeepDiver1975) [reply tests/env/start-swift-ceph.sh]: via the travis file you can control the vm which is used.

if it works on one setup but not on the other there is maybe a diff in the vm config.

please compare the yaml files

Comment (DeepDiver1975) [reply tests/env/start-swift-ceph.sh]: bildschirmfoto von 2017-08-10 16-52-37

why move to sudo: false? does this work together with service: docker? afaik not

Comment (mrow4a) [reply tests/env/start-swift-ceph.sh]: @DeepDiver1975 any example of vm selections? About the `sudo`, I tried to match objectstore.

**Comment by DeepDiver1975** [comment]
please give it a shot with a different docker image - see https://github.com/owncloud/objectstore/wiki/Start-ceph-demo-for-local-testing-....

2nd pr please - thx

**Pull request closed without merging**